pkg/cloudsql/clients.go

```diff
@@ -0,0 +1,123 @@
+package cloudsql
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/sqladmin/v1"
+)
+
+type CloudMonitoringClient struct {
+	client *monitoring.MetricClient
+	ctx    context.Context
+}
+
+type SqlAdminClient struct {
+	client *sqladmin.Service
+}
+
+func NewSqlAdminClient(client *sqladmin.Service) SqlAdminClient {
+	return SqlAdminClient{
+		client: client,
+	}
+}
+
+func (client *SqlAdminClient) ApplyFlags(projectId string, databaseName string, newFlags []*sqladmin.DatabaseFlags) error {
+	// get current flag values (so we don't unset any that are already set!)
+	inst, err := client.client.Instances.Get(projectId, databaseName).Do()
+	if err != nil {
+		return fmt.Errorf("Failed to get instance settings! %v", err)
+	}
+
+	// convert those into a map for easier merging
+	flagValues := make(map[string]string, len(inst.Settings.DatabaseFlags))
+
+	for _, flag := range inst.Settings.DatabaseFlags {
+		fmt.Printf("Previously set database flag, %s: %s\n", flag.Name, flag.Value)
```

Comment:
Use logger is possible, this seems like a debug statement

Reply:
I think this can just go - it was only for while I was developing